packages/react/src/ErrorBoundary.tsx

```diff
@@ -101,14 +124,56 @@ class BaseErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState
     }
 
     let childrenOrFallback = children
+
     if (this.state.isError) {
+      let isReThrow = false
+      if (throwOn) {
+        for (let i = 0; i < throwOn.length; i++) {
+          const error = throwOn[i],
+            condition = throwOn[i]
+          if (error.prototype instanceof Error) {
+            if (this.state.error instanceof error) {
+              isReThrow = true
+              break
+            }
+          } else if (typeof condition === 'function') {
+            if ((condition as Condition)(this.state.error)) {
+              isReThrow = true
+              break
+            }
+          }
```

Comment:
I'm not sure if it's exactly right. The opinion was that it would be easier to read and understand the code if each error throws a throw, as shown in the code below, and were divided into blocks or functions to determine the conditions.

```typescript
const isMatchWith = (errors) => errors
  .filter((error) => error.prototype instanceof Error)
  .some((error) => this.state.error instanceof error || (typeof error === 'function' && condition(this.state.error)))

const isMatchThrowOn = isMatchWith(throwOn)
const isMatchCatchOn = isMatchWith(catchOn)

const isReThrow = isMatchThrowOn === true && isMatchCatchOn === false;

if (isReThrow) {
  throw this.state.error;
}
```